lib/tango_client.rb

```diff
@@ -0,0 +1,90 @@
+require "httparty"
+require Rails.root.join("config", "autogradeConfig.rb")
+
+##
+# Ruby API Client of Tango
+module TangoClient
+  # Httparty client for Tango API
+  class ClientObj
+    include HTTParty
+    base_uri "#{RESTFUL_HOST}:#{RESTFUL_PORT}"
+    default_timeout 10
+  end
+
+  # Exception for Tango API Client
+  class TangoException < StandardError; end
+
+  def self.tango_handle_exceptions
+    resp = yield
+    if resp.content_type == "application/json" && resp["statusId"] && resp["statusId"] < 0
+      fail TangoException, "Tango returned negative status code."
+    end
+    return resp
+  rescue Net::OpenTimeout, Net::ReadTimeout
```

Comment:
Is there a set timeout limit? Or does this occur when we take Tango down and then try to hit it?